api/v1alpha1/mcpserver_types.go

```diff
@@ -146,6 +146,19 @@ type MCPServerSpec struct {
 
 	// HTTPTransport defines the configuration for a Streamable HTTP transport.
 	HTTPTransport *HTTPTransport `json:"httpTransport,omitempty"`
+
+	// Authentication defines the authentication configuration for the MCP server.
+	// This field is optional and can be used to configure JWT authentication.
+	// If not specified, the MCP server will not require authentication.
+	// +optional
+	Authentication *MCPServerAuthentication `json:"authentication,omitempty"`
+
+	// Authorization defines the authorization configuration for the MCP server.
+	// This field is optional and can be used to configure authorization rules
+	// for the MCP server. If not specified, the MCP server will not enforce
+	// any authorization rules.
+	// +optional
+	Authorization *MCPServerAuthorization `json:"authorization,omitempty"`
```

Comment:
Maybe this is dumb, but both of these words are relatively hard to spell. What do you think about changing these to `authn/authz`?